### ISSUE TITLE
validateCommand() fails if CWD is set and process is ran in a different directory

### ISSUE THREAD

**Issue by flaviovs** [ptlis/shell-command#4]
Value set using `setCwd()` is not used in `validateCommand()`, so validation fails if relative  command is not run in the same working directory.

How to test:

```php
<?php
require __DIR__ . '/vendor/autoload.php';
$builder = (new \ptlis\ShellCommand\CommandBuilder())
        ->setCwd('/')
        ->setCommand('./bin/ls');
echo $builder->buildCommand();
```

This will succeed if called in `/`, but fail if script is called from any other directory:

```console
PHP Fatal error:  Uncaught RuntimeException: Invalid command "./bin/ls" provided to ptlis\ShellCommand\CommandBuilder. in /app/vendor/ptlis/shell-command/src/CommandBuilder.php:180
```

Bug is in line 179 of `CommandBuilder->buildCommand()`:

https://github.com/ptlis/shell-command/blob/5835e2dfecf5cfab6c2c040bf5d921967e2ddf9d/src/CommandBuilder.php#L177-L181

Hand-made patch:

```
-         if (!$this->environment->validateCommand($this->command) { 
+         if (!$this->environment->validateCommand($this->command, $this->cwd)) { 
```

**Comment by ptlis** [comment]
Hi flaviovs, thanks for the bug report.

I attempted to address this last weekend (before I went on holiday) and published 0.17.2 which fixes this issue, but on doing so I found the travis builds were failing and I didn't have the time to get to the bottom of it.

I've taken another quick look this evening and I've been able to verify that this appears to be a bug with Travis' PHP builds, you can see my bug report [here](https://travis-ci.community/t/failures-when-using-chdir-and-proc-open/4131).

I'd recommend checking out a copy of the master branch & running the tests on the environment you're working on to make sure you've not got a similar issue; if all is fine then ^0.17.2 should have this fix.

**Comment by ptlis** [comment]
I've reworked the code that was failing on Travis to use a different approach and the build is now passing; please can you try again with release 0.17.3 and confirm whether this fix is working for you.
